### DISCUSSION TITLE
Http Upload as transwrap output

### DISCUSSION THREAD

**Discussion by donkon** [bbc/bmx#98] [General]
Is there a way to send the output of bmxtranswrap as an HTTP PUT request to a webserver by specifying something like `-method PUT -o <url>`? If not, could it be implemented?

**Comment by philipnbbc** [comment]
Only reading is currently implemented and I don't currently have plans to implement writing.

I imagine it would involve implementing the write methods in [MXFHTTPFile](https://github.com/bbc/bmx/blob/main/src/common/MXFHTTPFile.cpp). A simple implementation would support a chunked upload using PUT, which would limit support for a single pass MXF file write (e.g. duration is not set in the header). If the server supported writing byte ranges then that would support a multi-pass MXF file write. If the endpoint is an object store like S3 then multipart uploads could be used for a multi-pass MXF file write, with possible limitations like only update the header and footer partitions.
